kore/src/Kore/OnePath/Verification.hs

```diff
@@ -113,14 +114,22 @@ decision is subject to change without notice.
 
  -}
 
-{- | Wrapper for a rewrite rule that should be used as a claim.
+{- | Class for claim-like rules
 -}
-newtype Claim level = Claim
-    { unClaim :: RewriteRule level Variable
-    }
+class Claim claim where
```

Comment:
I don't like making classes that expose the underlying representation like this, because we can never change the representation. The only types that can correctly implement this class are `newtype` wrappers around `RewriteRule`; why not use [`Coercible`](https://hackage.haskell.org/package/base-4.12.0.0/docs/Data-Coerce.html#t:Coercible) instead?

By the way, because `RewriteRule`, `OnePathRule`, and `AllPathRule` each have a distinct Kore representation, I think we should use `RulePattern` instead of `RewriteRule` here.